security-policies/bundle/compliance/cis_azure/rules/cis_3_11/rule.rego

```diff
@@ -0,0 +1,31 @@
+package compliance.cis_azure.rules.cis_3_11
+
+import data.compliance.lib.common
+import data.compliance.policy.azure.data_adapter
+import future.keywords.if
+
+finding = result if {
+	# filter
+	data_adapter.is_storage_account
+	_ = data_adapter.resource.extension.blobService
+
+	# set result
+	result := common.generate_result_without_expected(
+		common.calculate_result(ensure_soft_delete_is_enabled),
+		{"Resource": data_adapter.resource},
+	)
+}
+
+default ensure_soft_delete_is_enabled = false
+
+ensure_soft_delete_is_enabled if {
+	is_policy_valid(data_adapter.resource.extension.blobService.deleteRetentionPolicy)
+	data_adapter.resource.extension.blobService.deleteRetentionPolicy.allowPermanentDelete == false
+	is_policy_valid(data_adapter.resource.extension.blobService.containerDeleteRetentionPolicy)
```

Comment:
Don't this need a check `allowPermanentDelete == false` too? Or whatever the counterpart in code of `enable-container-delete-retention is 'true'` is

Reply:
Indeed but what I observed from the API calls was that the container delete retention has no `allowPermanentDelete` field. I am not sure why. 

I mostly got data like this:

```json
      "containerDeleteRetentionPolicy": {
        "days": 7,
        "enabled": true
      },
      "deleteRetentionPolicy": {
        "allowPermanentDelete": false,
        "days": 7,
        "enabled": true
      },
```

So I wasn't confident in putting the `allowPermanentDelete` check for container as well.

Reply:
Hm. Maybe good to dig a bit deeper to understand this, because the whole point is to see if there is soft delete or not, right?

Atm containers are not checked for soft delete

Reply:
From what I saw by digging around the settings when the `soft delete` setting is not checked, the `containerDeleteRetentionPolicy` is missing entirely as a field. And when it is checked, then it is present.

It's the other way around. I think the  `allowPermanentDelete` check is wrong. 


<img width="385" alt="Screenshot 2023-12-21 at 12 24 17 PM" src="https://github.com/elastic/cloudbeat/assets/1380039/f718a5d4-6d7c-454b-9fce-7f05eae7de43">
It seems this is the checkbox `Enable permanent delete for soft deleted items` that says `Permanent delete enables you to permanently delete a soft-deleted snapshot or blob version before the retention period ends.` which is a different thing than having enabled the soft delete in general. 

That's why all these 3 checkboxes can be enabled simultaneously. 

I will remove the check on `allowPermanentDelete` from rego